Sources/ParseSwift/Authentication/3rd Party/ParseTwitter.swift

```diff
@@ -0,0 +1,267 @@
+//
+//  ParseTwitter.swift
+//  ParseSwift
+//
+//  Created by Abdulaziz Alhomaidhi on 3/17/21.
+//  Copyright © 2021 Parse Community. All rights reserved.
+//
+
+import Foundation
+#if canImport(Combine)
+import Combine
+#endif
+
+// swiftlint:disable line_length
+
+/**
+ Provides utility functions for working with Twitter User Authentication and `ParseUser`'s.
+ Be sure your Parse Server is configured for [sign in with Twitter](https://docs.parseplatform.org/parse-server/guide/#configuring-parse-server-for-sign-in-with-twitter).
+ For information on acquiring Twitter sign-in credentials to use with `ParseTwitter`, refer to [Twitter's Documentation](https://developer.twitter.com/en/docs/authentication/guides/log-in-with-twitter.
+ */
+public struct ParseTwitter<AuthenticatedUser: ParseUser>: ParseAuthentication {
+
+    /// Authentication keys required for Twitter authentication.
+    enum AuthenticationKeys: String, Codable {
+        case id // swiftlint:disable:this identifier_name
+        case token
```

Comment:
The keys for twitter don't seem correct, the [documentation]() shows:

```javascript
"twitter": {
    "id": "user's Twitter id number as a string",
    "consumer_key": "your application's consumer key",
    "consumer_secret": "your application's consumer secret",
    "auth_token": "an authorized Twitter token for the user with your application",
    "auth_token_secret": "the secret associated with the auth_token"
  }
``` 

which will need coding keys:

```swift
enum CodingKeys: String, CodingKey { 
  case id // swiftlint:disable:this identifier_name 
  case consumer_key = "consumer_key"
  case consumer_secret = "consumer_secret"
  case authToken = "auth_token"
  case authTokenSecret = "auth_token_secret"  
}
```

Reply:
done